addons/mail/tests/test_mail_activity.py

```diff
@@ -7,7 +7,7 @@
 from odoo import fields
 from odoo.addons.mail.models.mail_activity import MailActivity
 from odoo.addons.mail.tests.common import MailCommon
-from odoo.tests.common import Form, tagged, HttpCase
+from odoo.tests import Form, tagged, HttpCase
```

Comment:
Hello, just curious, is there any particular reason to import from `odoo.tests` rather than `odoo.tests.common`?

If the `Form` import is undesired, shouldn't it be on its own line, keeping other imports unchanged?

Is that something we should try to do generally speaking? Trying to import from the top-most possible place? What is the limit?

Reply:
Hi,

> Hello, just curious, is there any particular reason to import from odoo.tests rather than odoo.tests.common?

Not really, I just changed it for simplicity. The behavior remains the same thanks to https://github.com/odoo/odoo/blob/91b3813550234fef5508586151a95f34a1eabd8d/odoo/tests/__init__.py#L9

> Is that something we should try to do generally speaking? Trying to import from the top-most possible place?

I don't think so, but it's still more robust (import at top-most place) if we change the internal structure of the 'tests' module files later. But I don't think this is important.

Reply:
Would it make sense to make the `Form` object accessible on the `TransactionCase` to avoid any import at all?